elasticapm/utils/__init__.py

```diff
@@ -103,7 +103,10 @@ def is_master_process():
 
 def get_url_dict(url):
     scheme, netloc, path, params, query, fragment = compat.urlparse.urlparse(url)
-    if ":" in netloc:
+    if '[' in netloc and ']' in netloc:
```

Comment:
I had a quick look at `ParseResult` (the object that `urlparse` returns). It has properties for `hostname` and `port`, which properly support IPv6 URLs since Python 2.7. Only drawback I see is that it'll lowercase IPv6 addresses, but that seems like it isn't a huge problem.

So we would do something like this:

    parse_result = compat.urlparse.urlparse(url)

And then use `parse_result.hostname`, `parse_result.port`, `parse_result.scheme` etc. What do you think?

Reply:
@beniwohli You're right. I refactored the code to use `ParseResult`. Since the port is returned as integer, I had to add small hack to map into string, so not to introduce any regressions. Thanks for the help.